mlir/include/mlir/Dialect/Rock/Tuning/Serializable.h

```diff
@@ -52,15 +62,50 @@ template <class Derived, char Seperator = ','> struct Serializable {
     }
   };
   void serialize(std::ostream &stream) const {
+    stream << "v" << static_cast<int32_t>(version) << ":";
     char sep = 0;
     Derived::visit(static_cast<const Derived &>(*this),
                    std::bind(SerializeField{}, std::ref(stream), std::ref(sep),
                              std::placeholders::_1));
   }
 
-  bool deserialize(const std::string &s) {
+  bool checkVersionFormat(const std::string &s) {
+    const int32_t maxNumTokens = version == Version::V1 ? 8 : 9;
+    const int32_t maxNumSeperators = maxNumTokens - 1;
+    const int32_t minNumSeperators = maxNumSeperators - 2;
+    const auto numFoundSeperators = std::count_if(
+        s.begin(), s.end(), [](char c) { return c == Seperator; });
+    return numFoundSeperators >= minNumSeperators &&
+           numFoundSeperators <= maxNumSeperators;
+  }
+
+  bool deserialize(std::string s) {
+    llvm::Regex versionExpr("^v([0-9]+):");
+    llvm::SmallVector<llvm::StringRef, 2> matches;
+    if (versionExpr.match(s, &matches)) {
+      assert(matches.size() == 2 &&
+             "a match of the version regex expected 2 items");
+      int32_t value = std::stoi(matches[1].str());
+      if (value >= static_cast<int32_t>(Version::V1) &&
+          value < static_cast<int32_t>(Version::Count)) {
+        version = static_cast<Version>(value);
+        s = std::string(s.begin() + matches[0].size(), s.end());
+      } else {
+        // unknown perf config version
+        return false;
+      }
+    } else {
+      version = Version::V1;
+    }
+
+    if (!checkVersionFormat(s)) {
+      // incorrect perf config format
+      return false;
+    }
+
     auto out = static_cast<const Derived &>(*this);
-    bool ok = true;
+
+    bool ok{true};
```

Comment:
Can you put this back into `bool ok = true` since that's how these get written around here?

Reply:
Done